### PR TITLE
couchbase-lite-C 3.2.1

### PR DESCRIPTION
**Pull request by Nao-ris**
This pull request upgrades the repo to `couchbase-lite-C` 3.2.1.

> [!WARNING]  
> This release is not backward compatible with `couchbase-lite-C` 3.1.

Mainly, this new versions adds support for [Array Indexes](https://docs.couchbase.com/couchbase-lite/current/c/indexing.html#array-indexing).

### CODE REVIEW

File: src/index.rs [hunk 1]
```diff
@@ -35,6 +41,84 @@ impl ValueIndexConfiguration {
...
+    in documents, intended for use with the UNNEST query.
+    @param query_langage  The language used in the expressions (Required).
+    @param path  Path to the array, which can be nested to be indexed (Required).
+        Use "[]" to represent a property that is an array of each nested array level.
+        For a single array or the last level array, the "[]" is optional. For instance,
+        use "contacts[].phones" to specify an array of phones within each contact.
+    @param expressions  Optional expressions representing the values within the array to be
+        indexed. The expressions could be specified in a JSON Array or in N1QL syntax
+        using comma delimiter. If the array specified by the path contains scalar values,
+        the expressions should be left unset or set to null. */
```
Comment (EliseChouleur) [inline src/index.rs:69]: With a small example for this parameter ? :)

File: src/index.rs [hunk 2]
```diff
@@ -35,6 +41,84 @@ impl ValueIndexConfiguration {
...
+
+    pub fn name(&self) -> String {
+        unsafe {
+            CBLQueryIndex_Name(self.get_ref())
+                .to_string()
+                .unwrap_or_default()
+        }
+    }
+
+    pub fn collection(&self) -> Collection {
```
Comment (EliseChouleur) [inline src/index.rs:117]: As you've added documentation on other public api, please add some here as well

File: src/index.rs [hunk 3]
```diff
@@ -90,6 +174,23 @@ impl Collection {
         failure(err)
     }
 
+    pub fn create_array_index(&self, name: &str, config: &ArrayIndexConfiguration) -> Result<bool> {
```
Comment (EliseChouleur) [inline src/index.rs:177]: Idem

**Review by TonyMasson — APPROVED** [review]

File: c_playground/main.c [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    printf(
+        "CBL %s: %s%s%s\n",
+        DOMAINS[domain],
+        LEVEL_PREFIX[level],
+        (char*) message.buf,
+        LEVEL_SUFFIX[level]
+    );
+}
+
+int main(void) {
```
[the remark this answers is not in the document]
Comment (Nao-ris) [reply c_playground/main.c:25]: Good idea

File: src/index.rs [hunk 5]
```diff
@@ -24,6 +26,10 @@ impl CblRef for ValueIndexConfiguration {
 }
 
 impl ValueIndexConfiguration {
+    /** Create a Value Index Configuration.
```
[the remark this answers is not in the document]
Comment (Nao-ris) [reply src/index.rs]: Yes, I reworked the documentation in the file, it was lacking.

**Pull request merged**